### DISCUSSION TITLE
Motion/Object Mask Not Working

### DISCUSSION THREAD

**Discussion by ScottulusMaximus** [blakeblackshear/frigate#14208] [General Support] [answered]
Hi all! Fantastic project, thank you!

![Screenshot 2024-10-08 095043](https://github.com/user-attachments/assets/19d665ab-14fc-4724-89a1-bb92a2dd19dc)

![Screenshot 2024-10-08 095133](https://github.com/user-attachments/assets/8fc775e9-0b84-4630-bc2b-9053be8b551e)

Having a bit of an issue though. There's a branch of a tree that when a car's headlight hits it I get a person flag. I've added both a motion and an object mask around it but it's still detecting an event. This then sets off my alarm and wakes the whole house up...

I'm running a Frigate+ model trained on nearly 800 images on a Coral. I've got a few of these false detections uploaded but I don't really want to request a new model (already have 3 used in 2 months) unless it will fix it. I can't get images of true detections there because it's in a tree.

Any ideas?

**Comment by lawul** [comment, marked as answer]
The masks only work if the bottom center of the bounding box is inside the masked area. (see https://docs.frigate.video/configuration/masks)
In your example, the bottom center is outside the masked area.

I'd propose that you pull the object mask further down but leave the motion mask as it is (for better/faster detection in case someone climbs over the fence).

**Comment by ScottulusMaximus** [comment]
Will give it a shot, thanks!

**Comment by ScottulusMaximus** [comment]
That worked thanks.

Any particular reason it's the bottom middle of the box though? My area is much bigger than it needs to be compared to if the detect point was in the centre of the bounding box.

**Reply by NickM-27** [reply]
So you can differentiate between an object standing closer to the camera vs father away. If it was the middle then an object far away or right up to the camera would both be considered in the zone which is wrong

**Reply by ScottulusMaximus** [reply]
From the proverbial horses mouth as it were, thanks for replying.

All sorted, hopefully the next training I'll be able to get rid of the object mask.
